pkg/ocp/ocp.go

```diff
@@ -19,62 +23,66 @@ type ConfigFile struct {
 	Content  []byte
 }
 
-type ConfigMaster struct {
+type OAuthConfig struct {
 	ConfigFile
-	OCP3 ocp3.Master
-	OCP4 ocp4.Master
+	OCP3    configv1.MasterConfig
+	OAuth   oauth.OAuthCRD
+	Secrets []secrets.Secret
 }
 
-type ConfigNode struct {
+type SDNConfig struct {
 	ConfigFile
-	OCP3 ocp3.Node
-	OCP4 ocp4.Node
+	OCP3 configv1.MasterNetworkConfig
+	SDN  sdn.NetworkCR
 }
 
 type Translator interface {
 	Add(string)
-	Decode()
-	Fetch(string)
+	Extract()
+	Transform()
 	GenYAML() ocp4.Manifests
-	Translate()
 }
 
 // GetFile allows to mock file retrieval
 var GetFile = io.GetFile
 
-func (config *ConfigMaster) Add(hostname string) {
-	masterf := env.Config().GetString("MasterConfigFile")
+var source = env.Config().GetString("Source")
 
-	if masterf == "" {
-		masterf = "/etc/origin/master/master-config.yaml"
+func (sdnConfig *SDNConfig) Add(hostname string) {
```

Comment:
It's not clear what the purpose of this method is, but from what I can see we're initializing input configs. IMO this should be done when the object is constructed.